### DISCUSSION TITLE
exception 처리를 어떻게 할 것인가?

### DISCUSSION THREAD

**Discussion by jurlring** [woowacourse-teams/2022-gong-seek#16] [Q&A]
# 표준 예외
평소에 쓰던 exception을 사용한다.
# 커스텀 예외
기능이랑 status code 별로 나눠서 exception을 작성한다.
## 커스텀 예외를 사용한다면, 계층 구조는 어떻게 가져갈 것인가?

**Comment by Gyuchool** [comment]
커스텀 예외를 하면 controller advice에서 예외 캐치하기 좋을 것 같아요. 비즈니스적 예외들만 처리하거나 비즈니스 외적 예외들로 계층 나눌수도 있다고 생각합니다. 그리고 커스텀을 안쓰면 예외 메세지만 보고 어떤 메세지인지 판단해야하는데 상수로 정의된 메세지만보고 해당 예외라는 것을 믿고 가기엔 변경 요소가 너무 많은 것 같습니다
